api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -73,7 +84,10 @@ public static TracerFactory getTracerFactory() {
    * @since 0.1.0
    */
   public static MeterFactory getMeterFactory() {
-    return getInstance().meterFactory;
+    if (instance.meterFactory.get() == null) {
+      setMeterFactory(SpiOpenTelemetryProvider.makeSpiMeterFactory());
```

Comment:
I was expecting to see some logic that uses a global "deferred" factory here. I see two cases:
(1) The user is planning to call setXXXerFactory() later; they want a handle they can store now which becomes functional later
(2) The user has planned for dependency injection, in which case we can bypass the user of a deferred factory here and use the injected value.
What I see here looks different. If the value is used it forces a decision about SPI to happen--what if there's no value injected and the user is still planning to call `setXXXFactory`? I don't see how that will work.
What I read in the code here is that if someone accesses the global

Reply:
This PR doesn't include having done anything with the Meter implementation, only Tracers.

Reply:
I see now. Inside the call to `makeSpiXXXFactory` you get a default "deferred" impl if there's no SPI value.